tcms/testruns/tests/test_views.py

```diff
@@ -629,3 +629,104 @@ def test_show_add_cases_to_run(self):
+
+    def test_get_remove_cases_from_run_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.delete_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.remove_cases_html, html=True)
+
+    def test_get_update_caserun_text_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.update_case_run_text_html, html=True)
+
+    def test_get_change_assignee_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.change_assignee_html, html=True)
+
+    def test_get_status_options_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+
+        for html_code in self.caserun_status_options_html:
+            self.assertContains(response, html_code, html=True)
+
+    def test_get_add_cases_to_run_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.add_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertNotContains(response, self.add_cases_html, html=True)
+
+    def test_get_remove_cases_from_run_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.delete_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertNotContains(response, self.remove_cases_html, html=True)
+
+    def test_get_update_caserun_text_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertNotContains(response, self.update_case_run_text_html, html=True)
+
+    def test_get_change_assignee_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertNotContains(response, self.change_assignee_html, html=True)
+
+    def test_get_status_options_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+
+        for tcrs in TestCaseRunStatus.objects.all():
```

Comment:
You have to iterate over the html menu options, not the status objects in the database.

Reply:
Isn't the purpose of these tests to check whether the menu options exist in the HTML at all?
From the perspective of the tests the HTML elements are not certain and so we need something else to compare them to.